.github/workflows/release-build-publish.yml

```diff
@@ -0,0 +1,256 @@
+name: Release Build Publish
+
+on:
+    release:
+      types: [ published ]
+
+jobs:
+  extract_project_name:
+    runs-on: ubuntu-latest
+    outputs:
+      cleanlab_package_name: ${{ steps.extract_name.outputs.value }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: TOML Reader
+        uses: SebRollen/toml-action@v1.2.0
+        id: extract_name
+        with:
+          file: "pyproject.toml"
+          field: "project.name"
+
+  store_published_version:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.compare.outputs.version }}
+    steps:
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install package
+        run: pip install .
+
+      - name: Compare the published version with the package version
+        id: compare
+        run: |
+          TAG_NAME="${{ github.ref_name }}"
+          # TAG_NAME must be in the form "v1.2.3[...]", error if it doesn't match
+            if [[ ! $TAG_NAME =~ ^v[0-9]+\.[0-9]+\.[0-9]+ ]]; then
+                echo "Tag name '$TAG_NAME' does not match the expected pattern 'vX.Y.Z'"
+                exit 1
+            fi
+          # Strip the 'v' prefix from the tag, and store the version in an output variable
+          VERSION=${TAG_NAME:1} # Strip the 'v' prefix from the tag
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+          # Check that the package version matches the tag version
+          PACKAGE_VERSION=$(python -c "import cleanlab; print(cleanlab.__version__)")
+          if [ "$PACKAGE_VERSION" != "$VERSION" ]; then
+            echo "Package version $PACKAGE_VERSION does not match tag $VERSION"
+            exit 1
+          fi
```

Comment:
For clarity.

The goal of this job is two-fold.

- Ensure that the setup.py still works by ensuring that `import cleanlab` properly sets up the necessary variables. That's why the `cleanlab.__version__` variable is still passed to `setuptools.setup()` in the setup.py file.
- The release tag should match the `cleanlab.__version__` that will be uploaded to PyPI.